--- ao/ao_private.h	2012-02-14 09:46:05.000000000 +0900
+++ ao/ao_private.h	2017-02-03 22:32:11.821947385 +0900
@@ -138,13 +138,13 @@
 
 void ao_read_config_files (ao_config *config);
 
-#define adebug(format, args...) {\
+#define adebug(format, ...) {\
     if(device->verbose==2){                                             \
       if(strcmp(format,"\n")){                                          \
         if(device->funcs->driver_info()->short_name){                   \
-          fprintf(stderr,"ao_%s debug: " format,device->funcs->driver_info()->short_name,## args); \
+          fprintf(stderr,"ao_%s debug: " format,device->funcs->driver_info()->short_name,__VA_ARGS__); \
         }else{                                                          \
-          fprintf(stderr,"debug: " format,## args);                     \
+          fprintf(stderr,"debug: " format,__VA_ARGS__);                 \
         }                                                               \
       }else{                                                            \
         fprintf(stderr,"\n");                                           \
@@ -152,13 +152,13 @@
     }                                                                   \
   }
 
-#define averbose(format, args...) {\
+#define averbose(format, ...) {\
     if(device->verbose>0){                                              \
       if(strcmp(format,"\n")){                                          \
         if(device->funcs->driver_info()->short_name){                   \
-          fprintf(stderr,"ao_%s info: " format,device->funcs->driver_info()->short_name,## args); \
+          fprintf(stderr,"ao_%s info: " format,device->funcs->driver_info()->short_name,__VA_ARGS__); \
         }else{                                                          \
-          fprintf(stderr,"info: " format,## args);                      \
+          fprintf(stderr,"info: " format,__VA_ARGS__);                  \
         }                                                               \
       }else{                                                            \
         fprintf(stderr,"\n");                                           \
@@ -166,13 +166,13 @@
     }                                                                   \
   }
 
-#define ainfo(format, args...) {\
+#define ainfo(format, ...) {\
     if(device->verbose>=0){                                             \
       if(strcmp(format,"\n")){                                          \
         if(device->funcs->driver_info()->short_name){                   \
-          fprintf(stderr,"ao_%s info: " format,device->funcs->driver_info()->short_name,## args); \
+          fprintf(stderr,"ao_%s info: " format,device->funcs->driver_info()->short_name,__VA_ARGS__); \
         }else{                                                          \
-          fprintf(stderr,"info: " format,## args);                      \
+          fprintf(stderr,"info: " format,__VA_ARGS__);                  \
         }                                                               \
       }else{                                                            \
         fprintf(stderr,"\n");                                           \
@@ -180,13 +180,13 @@
     }                                                                   \
   }
 
-#define awarn(format, args...) {\
+#define awarn(format, ...) {\
     if(device->verbose>=0){                                             \
       if(strcmp(format,"\n")){                                          \
         if(device->funcs->driver_info()->short_name){                   \
-          fprintf(stderr,"ao_%s WARNING: " format,device->funcs->driver_info()->short_name,## args); \
+          fprintf(stderr,"ao_%s WARNING: " format,device->funcs->driver_info()->short_name,__VA_ARGS__); \
         }else{                                                          \
-          fprintf(stderr,"WARNING: " format,## args);                   \
+          fprintf(stderr,"WARNING: " format,__VA_ARGS__);               \
         }                                                               \
       }else{                                                            \
         fprintf(stderr,"\n");                                           \
@@ -194,13 +194,13 @@
     }                                                                   \
   }
 
-#define aerror(format, args...) {                                       \
+#define aerror(format, ...) {\
     if(device->verbose>=0){                                             \
       if(strcmp(format,"\n")){                                          \
         if(device->funcs->driver_info()->short_name){                   \
-          fprintf(stderr,"ao_%s ERROR: " format,device->funcs->driver_info()->short_name,## args); \
+          fprintf(stderr,"ao_%s ERROR: " format,device->funcs->driver_info()->short_name,__VA_ARGS__); \
         }else{                                                          \
-          fprintf(stderr,"ERROR: " format,## args);                     \
+          fprintf(stderr,"ERROR: " format,__VA_ARGS__);                 \
         }                                                               \
       }else{                                                            \
         fprintf(stderr,"\n");                                           \
